code/modules/admin/admin_verbs.dm

```diff
@@ -322,10 +322,9 @@ var/list/admin_verbs_hideable = list(
 	if(istype(mob,/mob/dead/observer))
 		//re-enter
 		var/mob/dead/observer/ghost = mob
-		if (!ghost.can_reenter_corpse)
+		if (ghost.can_reenter_corpse)
 			log_admin("[key_name(usr)] re-entered corpse")
 			message_admins("[key_name_admin(usr)] re-entered corpse")
-		ghost.can_reenter_corpse = 1			//just in-case.
```

Comment:
The thing is, the normal ghosting will take your ckey out of the mob, aghosting won't. If you re-create a body for you, the aghosting will work normally. And even without the change I'm pretty sure the aghost won't work for normal ghosting